<launch>
    <!-- Localization-only mode -->
    <arg name="localization"            default="false"/>

    <arg     if="$(arg localization)" name="args" default=""/>
    <arg unless="$(arg localization)" name="args" default="--delete_db_on_start"/>

    <!-- RGB-D related topics -->
    <arg name="camera_name"             default="airsim_ros_node/Copter" />
    <arg name="rgb_topic"               default="/$(arg camera_name)/front_left_custom/Scene" />
    <arg name="depth_topic"             default="/$(arg camera_name)/front_left_custom/DepthPlanar" />
    <arg name="camera_info_topic"       default="/$(arg camera_name)/front_left_custom/Scene/camera_info" />
    <arg name="depth_camera_info_topic" default="/$(arg camera_name)/front_left_custom/DepthPlanar/camera_info" />

    <arg name="odom_topic"              default="/mavros/local_position/odom" />    <!-- Odometry topic name -->

    <!-- Visual Odometry -->
    <!--
    <node name="rtabmap_stereo_odometry" pkg="rtabmap_ros" type="stereo_odometry" output="screen" args="" launch-prefix="">    
        <remap from="/left/image_rect"      to="$(arg camera_name)/zed_node/left/image_rect_color"/>
        <remap from="/right/image_rect"     to="$(arg camera_name)/zed_node/right/image_rect_color"/>
        <remap from="/left/camera_info"     to="$(arg camera_name)/zed_node/left/camera_info"/>
        <remap from="/right/camera_info"    to="$(arg camera_name)/zed_node/right/camera_info"/>
    </node>
    -->

    <!-- RTABmapviz -->
    <node name="rtabmapviz" pkg="rtabmap_ros" type="rtabmapviz" output="screen" args="" launch-prefix="">    
        <remap from="rgb/image"         to="$(arg rgb_topic)"/>
        <remap from="depth/image"       to="$(arg depth_topic)"/>
        <remap from="rgb/camera_info"   to="$(arg camera_info_topic)"/>

        <remap from="grid_map"          to="map" />
        <remap from="odom"              to="$(arg odom_topic)"/>
    </node> 


    <!-- RTABmap -->
    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="log" args="$(arg args)" launch-prefix="">
        <rosparam command="load" file="$(find airsim_launch)/config/rtabmap.yaml" />
    
        <remap from="rgb/image"         to="$(arg rgb_topic)"/>
        <remap from="depth/image"       to="$(arg depth_topic)"/>
        <remap from="rgb/camera_info"   to="$(arg camera_info_topic)"/>

        <remap from="grid_map"          to="map" />
        <!-- <remap from="odom"              to="$(arg odom_topic)"/> -->

        <!-- localization mode -->
        <param     if="$(arg localization)" name="Mem/IncrementalMemory" type="string" value="false"/>
        <param unless="$(arg localization)" name="Mem/IncrementalMemory" type="string" value="true"/>
        <param name="Mem/InitWMWithAllNodes" type="string" value="$(arg localization)"/>
    </node>

</launch>

